sklearn/neural_network/_base.py

```diff
@@ -166,15 +166,23 @@ def squared_loss(y_true, y_pred):
     y_pred : array-like or label indicator matrix
         Predicted values, as returned by a regression estimator.
 
+    sample_weight: array-like of shape (n_samples,)
+        sample weights.
+
     Returns
     -------
     loss : float
         The degree to which the samples are correctly predicted.
     """
-    return ((y_true - y_pred) ** 2).mean() / 2
+    loss = ((y_true - y_pred) ** 2).sum(axis=-1)
+
+    if sample_weight is not None:
+        loss = loss * sample_weight
+
+    return loss.mean() / 2
```

Comment:
When `sample_weight=None`, the operation becomes:

```python
loss = ((y_true - y_pred) ** 2).sum(axis=-1)
return loss.mean() / 2
```

I do not think that is the same as `((y_true - y_pred) ** 2).mean() / 2`